experimental/StandardFiniteFields/StandardFiniteFields.jl

```diff
@@ -0,0 +1,462 @@
+    spdr = get!(spd, r, Dict{ZZRingElem, ZZRingElem})
+    spdr[k] = nr
+end
+function set_steinitz_prime_degree!(f::Function, F::FinField, r::IntegerUnion, k::IntegerUnion)
+    spd = get_attribute!(F, :steinitz_prime_degree, Dict{ZZRingElem, Dict{ZZRingElem, ZZRingElem}}())
+    spdr = get!(spd, r, Dict{ZZRingElem, ZZRingElem})
+    spd[r][k] = f()
+end
+function set_standard_extension!(F::PrimeField, n::IntegerUnion, K::FinField)
+    ext = get_attribute!(F, :standard_extensions, Dict{ZZRingElem, FinField}())
+    ext[n] = K
+end
+
+
+function is_standard_prime_field(F::PrimeField)
+    get_attribute(F, :is_standard_prime_field, false)
+end
+function is_standard_finite_field(F::FinField)
+    get_attribute(F, :is_standard_finite_field, false)
+end
+function primitive_powers_in_tower_basis(F::FinField)
+    get_attribute(F, :primitive_powers_in_tower_basis, nothing)
+end
+function tower_basis(F::FinField)
+    get_attribute(F, :tower_basis, nothing)
+end
+function get_steinitz_prime_degree(F::FinField, r::IntegerUnion, k::IntegerUnion)
+    spd = get_attribute(F, :steinitz_prime_degree, nothing)
+    spdr = get(spd, r, nothing)
+    get(spdr, k, nothing)
+end
+function get_steinitz_prime_degree!(f::Function, F::FinField, r::IntegerUnion, k::IntegerUnion)
+    spd = get_attribute!(F, :steinitz_prime_degree, Dict{ZZRingElem, Dict{ZZRingElem, ZZRingElem}}() )
+    spdr = get!(spd, r, Dict{ZZRingElem, ZZRingElem}())
+    get!(spdr, k, f())
+end
+function get_standard_extensions(F::PrimeField)
+  get_attribute(F, :standard_extensions, nothing)
+end
+function get_standard_extensions!(F::PrimeField)
+  get_attribute(F, :standard_extensions, Dict{ZZRingElem, FinField}())
+end
+function get_standard_extension(F::PrimeField, k::IntegerUnion)
+  ext = get_attribute(F, :standard_extensions, nothing)
+  get(ext, k, nothing)
+end
+function get_standard_extension!(F::PrimeField, k::IntegerUnion, L::FinField)
+  ext = get_attribute!(F, :standard_extensions, Dict{ZZRingElem, FinField}())
+  get!(ext, k, L)
+end
+function get_standard_extension!(f::Function, F::PrimeField, k::IntegerUnion)
+  ext = get_attribute!(F, :standard_extensions, Dict{ZZRingElem, FinField}())
+  get!(f, ext, k)
+end
+
+
+# TODO: Lübeck speeds this up by caching triples [q,m,a] resulting from this
+function standard_affine_shift(q::IntegerUnion, i::IntegerUnion)
+    m = div(4*q, 5)
+    while gcd(m,q) != 1
+        m -= 1
+    end
+    a = div(2*q, 3)
+    return mod((m*i + a), q)
+end
+
+# Given a field F and Steinitz number n, give the corresponding field element.
+# we REQUIRE that F is a standard finite field TODO: using @assert?
+function element_from_steinitz_number(F::PrimeField, n::IntegerUnion)
+    return F(n)
+end
+function element_from_steinitz_number(F::FinField, n::IntegerUnion)
+    p = characteristic(F)
+    q = order(F)
+    if n < 0 || n > q
+        error("We need to have 0 <= n <= q")
+    end
+    if n == 0
+        return zero(F)
+    else
```

Comment:
More idiomatic would be
```suggestion
    n == 0 && return zero(F)
```
and then unindent the following code